x-pack/plugin/ml-feature-index-builder/src/test/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/AbstractSerializingFeatureIndexBuilderTestCase.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.featureindexbuilder.job;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+import org.junit.Before;
+
+import static java.util.Collections.emptyList;
+
+public abstract class AbstractSerializingFeatureIndexBuilderTestCase<T extends ToXContent & Writeable>
+        extends AbstractSerializingTestCase<T> {
+
+    /**
+     * Test case that ensure aggregation named objects are registered
+     */
+
+    private NamedWriteableRegistry namedWriteableRegistry;
+    private NamedXContentRegistry namedXContentRegistry;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
```

Comment:
It's unusual to override the base class `@Before` method and then call the base class version first.  JUnit will by default call all the base class `@Before` methods before the derived class `@Before` methods, so it would be more usual to just give the derived class `@Before` method a different name and let JUnit call the two methods in the standard order, avoiding the need to call `super.setUp()` here.